### DISCUSSION TITLE
Dyson Purifier Big+Quiet Formaldehyde BP04

### DISCUSSION THREAD

**Discussion by Anto79-ops** [libdyson-wg/ha-dyson#63] [General]
I'm thinking about getting the Dyson Purifier Big+Quiet Formaldehyde BP04

Think it will be hard to integrate? I'm open to working with you to figure out how. 

Cheers

**Comment by dotvezz** [comment]
It shouldn't be that hard! Matter of fact, I've already got an MQTT dump from another user with the Non-Formaldehyde Big+Quiet and it's _very_ similar to other devices. If you do get it, feel free to do the same; post the results of get_devices.py and [dyson-mqtt-listen](https://github.com/dotvezz/dyson-mqtt-listen/releases/tag/v1.0.0).

I'm quietly working on a refactor for universal support, which is made possible by the merge with the cloud integration. That way I won't need to stay on top of new products (until Dyson decides to change all their mqtt message keys lol).

**Reply by dotvezz** [reply]
Actually I lied, the other user provided a dump for the Big+Quiet Formaldehyde - so don't feel any pressure to provide mqtt messages for me lol. I won't complain if you _do_ though.

**Reply by Anto79-ops** [reply]
OK, that sounds good!

When I do get it, I will give it a go.

Cheers

**Reply by dotvezz** [reply]
Just released [v1.1.0](https://github.com/libdyson-wg/ha-dyson/releases/tag/v1.1.0) with initial support for Big+Quiet. If you wind up getting one, check it out and let me know if there seems to be anything missing.

**Reply by Anto79-ops** [reply]
Awesome 👌 

It might be a month or two, but when I do get it, I will follow up here. Thank you very much.

**Comment by ku1ik** [comment]
I have integrated this purifier today and it works like a charm :ok_hand: 

Thank you for your work on this project :heart:

**Comment by Anto79-ops** [comment]
looks like this is fully supported! thanks
